File: pizza-store-ui/src/app/core/models/order.ts
export interface OrderDetailRequest {
  productId?: number;
  comboId?: number;
  sizeId?: number;
  quantity: number;
  toppingIds?: number[];
}

export interface OrderRequest {
  userId?: number;
  customerName: string;
  customerPhone: string;
  shippingAddress: string;
  note?: string;
  paymentMethod: string;
  couponCode?: string;
  orderDetails: OrderDetailRequest[];
}

export enum OrderStatus {
  PENDING = 'PENDING',
  CONFIRMED = 'CONFIRMED',
  PREPARING = 'PREPARING',
  READY = 'READY',
  SHIPPING = 'SHIPPING',
  DELIVERED = 'DELIVERED',
  CANCELLED = 'CANCELLED'
}

export interface OrderToppingResponse {
  toppingId: number;
  name: string;
  price: number;
}

export interface OrderDetailResponse {
  id: number;
  orderId: number;
  productId?: number;
  productName?: string;
  productImageUrl?: string;
  sizeId?: number;
  sizeName?: string;
  comboId?: number;
  comboName?: string;
  comboImageUrl?: string;
  quantity: number;
  price: number;
  toppingDetails?: OrderToppingResponse[];
}

export interface OrderResponse {
  id: number;
  userId: number;
  customerName: string;
  customerPhone: string;
  shippingAddress: string;
  note?: string;
  totalPrice: number;
  status: OrderStatus;
  paymentStatus: string;
  paymentMethod: string;
  createdAt: string;
  updatedAt: string;
  couponId?: number;
  discountApplied?: number;
  orderDetails?: OrderDetailResponse[];
}

export interface PageResponse<T> {
  content: T[];
  currentPage: number;
  totalPages: number;
  totalElements: number;
}

File: pizza_store_api/src/main/java/fa/training/pizza_store_api/model/Order.java
package fa.training.pizza_store_api.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import fa.training.pizza_store_api.enums.OrderStatus;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Order {
    private Integer id;
    private Integer userId;
    private String customerName;
    private String customerPhone;
    private String shippingAddress;
    private String note;
    private BigDecimal totalPrice;
    private OrderStatus status;
    private String paymentStatus;
    private String paymentMethod;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    private Integer couponId;
    private BigDecimal discountApplied;

    private List<OrderDetail> orderDetails;
}
